src/randoop/ExceptionCheck.java

```diff
@@ -0,0 +1,145 @@
+ * differ on whether the expectation of the exception is enforced.
+ */
+public abstract class ExceptionCheck implements Check {
+  
+ /**
+   * 
+   */
+  private static final long serialVersionUID = 4806179088639914364L;
+
+  protected final Throwable exception;
+  
+  // Indicates which statement results in the given exception. 
+  protected final int statementIndex;
+
+  public ExceptionCheck(Throwable exception, int statementIndex) {
+    this.exception = exception;
+    this.statementIndex = statementIndex;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (o == null) return false;
+    if (o == this) return true;
+    if (this.getClass() != o.getClass()) { //match implementing class
+      return false;
+    }
+    ExceptionCheck other = (ExceptionCheck)o;
+    return this.exception.equals(other.exception) 
+        && statementIndex == other.statementIndex;
+  }
+  
+  @Override
+  public int hashCode() {
+    return Objects.hash(exception,statementIndex);
+  }
+  
+  @Override
+  public String toString() {
+    return "// throws exception of type " + exception.getClass().getCanonicalName() + Globals.lineSep;
+  }
+
+  /** Returns the class of the exception thrown**/
+  @Override
+  public String get_value() {
+    return exception.getClass().getName();
+  }
+
+  @Override
+  public String get_id() {
+    return "Throws exception @" + statementIndex;
+  }
+
+  /** return the offset of this statement in the sequence **/
+  @Override
+  public int get_stmt_no() {
+    return statementIndex;
+  }
+
+  /**
+   * The "try" half of the try-catch wrapper.
+   */
+  @Override
+  public String toCodeStringPreStatement() {
+    StringBuilder b = new StringBuilder();
+    b.append("// The following exception was thrown during execution." + Globals.lineSep);
+    b.append("// This behavior will recorded for regression testing." + Globals.lineSep);
+    b.append("try {" + Globals.lineSep + "  ");
+    return b.toString();
+  }
+  
+  /** The nearest visible superclass -- usually the argument itself. */
+  // TODO: handle general visibility based on Randoop command-line
+  // arguments, rather than hard-coding isPublic test.
+  public static Class<?> nearestVisibleSuperclass(Class<?> clazz) {
+    while (! Modifier.isPublic(clazz.getModifiers())) {
+      clazz = clazz.getSuperclass();
+    }
+    return clazz;
+  }
+
+  /**
+   * Returns the "catch" half of the try-catch wrapper.
+   * Catches the closest public superclass of the exception.
+   * Calls {@link this#appendCatchBehavior(StringBuilder, String)} to determine
+   * what code to include in the catch block.
+   */
+  @Override
+  public String toCodeStringPostStatement() {
+    StringBuilder b = new StringBuilder();
+    String exceptionClassName = getExceptionName();
+    if (exceptionClassName == null) {
+      exceptionClassName = "Exception";
+    }
+    appendTryBehavior(b, exceptionClassName);
+    if (Modifier.isPublic(exception.getClass().getModifiers())) {
```

Comment:
Why isn't appendCatchBehavior called in the then clause?  I expected the if...then...else to compute the type being caught, but then to re-merge so that appendCatchBehavior is always called.